modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtGetSingleFuture.java

```diff
@@ -311,7 +313,14 @@ private boolean map(KeyCacheObject key, boolean forceKeys) {
 
             // By reserving, we make sure that partition won't be unloaded while processed.
             if (part.reserve()) {
-                if (forceKeys || (part.state() == OWNING || part.state() == LOST)) {
+                boolean mvccNoOwners = false;
+
+                // Force key request is disabled for MVCC. So if there are no partition owners for the given key
+                // we need to set flag forceKeys to true to avoid useless remapping to other non-owning partitions.
+                if (cctx.mvccEnabled())
```

Comment:
Should partition state be LOST if it has no owners?
Is additional flag really needed here?